src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -218,18 +218,21 @@
 
   <!-- archive the test binaries along with some supporting files -->
   <Target Name="ArchiveTestBuild" AfterTargets="CreateAssemblyListTxt" Condition="'$(ArchiveTests)' == 'true'">
+    <PropertyGroup Condition="'$(TestProjectName)'==''">
+      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+    </PropertyGroup>
     <PropertyGroup>
       <TestArchiveDir>$(TestWorkingDir)$(OSPlatformConfig)/archive/tests/</TestArchiveDir>
       <ProjectJson Condition="!Exists('$(ProjectJson)')">$(OriginalProjectJson)</ProjectJson>
-    </PropertyGroup>
-    <PropertyGroup Condition="'$(TestProjectName)'==''">
-      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+      <DestinationArchiveFile>$(TestArchiveDir)$(TestProjectName)</DestinationArchiveFile>
+      <DestinationArchiveFile Condition="'$(TargetGroup)' != ''">$(DestinationArchiveFile).$(TargetGroup)</DestinationArchiveFile>
```

Comment:
Hmm yes this might cause a problem with Helix testing as we use file archive name minus the extension to determine the test binary name.  I think we'll need to place these archives in their own sub-directory.